engine/src/java/com/runrev/android/Engine.java

```diff
@@ -3153,15 +3153,20 @@ private void onMediaResult(int resultCode, Intent data)
         if (resultCode == Activity.RESULT_OK)
 		{
             Uri t_data = data.getData();
-            String t_path = null;
+            String t_path = "";
             if (t_data != null)
             {
-                Cursor t_cursor = ((LiveCodeActivity)getContext()).getContentResolver().query(t_data, null, null, null, null);
+            		Cursor t_cursor = null;
+                try
+				{
+					t_cursor = ((LiveCodeActivity) getContext()).getContentResolver().query(t_data, null, null, null, null);
+				}catch(SecurityException e){}
                 if (t_cursor != null)
                 {
                     t_cursor.moveToFirst();
                     int t_index = t_cursor.getColumnIndex (MediaStore.Images.ImageColumns.DATA);
-                    t_path = t_cursor.getString(t_index);
+                    if (t_index > 0)
+                    		t_path = t_cursor.getString(t_index);
```

Comment:
Hi Hedgehao,

I think you can avoid this check by rearranging some code into the `try` block:

```java
Cursor t_cursor = null;
try
{
    t_cursor = ((LiveCodeActivity) getContext()).getContentResolver().query(t_data, null, null, null, null);
    if (t_cursor != null)
    {
        t_cursor.moveToFirst();
        int t_index = t_cursor.getColumnIndex (MediaStore.Images.ImageColumns.DATA);
        t_path = t_cursor.getString(t_index);
    }
} catch (SecurityException e) {}
```